fix: LEAP-854: NER Annotations disappear from UI when switching between annotations

juliosgarbi opened the pull request:
### PR fulfills these requirements
- [x] Commit message(s) and PR title follows the format `[fix|feat|ci|chore|doc]: TICKET-ID: Short description of change made` ex. `fix: DEV-XXXX: Removed inconsistent code usage causing intermittent errors`
- [x] Tests for the changes have been added/updated (for bug fixes/features)
- [ ] Docs have been added/updated (for bug fixes/features)
- [x] Best efforts were made to ensure docs/code are concise and coherent (checked for spelling/grammatical errors, commented out code, debug logs etc.)
- [x] Self-reviewed and ran all changes on a local instance (for bug fixes/features)

This PR addresses the issue where NER annotations disappear from the UI when switching between annotations and predictions.

#### Change has impacts in these area(s)
_(check all that apply)_
- [ ] Product design
- [ ] Backend (Database)
- [ ] Backend (API)
- [x] Frontend



### Describe the reason for change
When the user is using the HyperText tag with the property `inline` set to false, the HyperText component loads content into an iframe. When the user switches between annotation tabs, the regions are not being cleared from memory. Consequently, the regions are not reloaded properly when switching back to annotations.

![Screen-Recording-2024-05-22-at-12 20 29](https://github.com/HumanSignal/label-studio/assets/4741623/3f223dc3-a3b2-4baf-8091-eb4cc1931a93)

#### What does this fix?
The fix involves validating the component within the componentWillUnmount method and disposing of the spans. This process cleans the variable, ensuring that self._spans is empty each time the component is loaded, allowing the regions to be reloaded correctly.

![Screen-Recording-2024-05-22-at-12 21 23](https://github.com/HumanSignal/label-studio/assets/4741623/00cd9afa-5b61-4b0a-a0ef-e82cb617f83c)

### What level of testing was included in the change?
_(check all that apply)_
- [x] e2e
- [ ] integration
- [ ] unit

juliosgarbi committed:
fix: LEAP-854: NER Annotations disappear from UI when switching between annotations

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/9194106349

juliosgarbi committed:
fix biome tests

juliosgarbi committed:
change test name

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/9194202980

hlomzik commented on web/libs/editor/src/tags/object/RichText/model.js:
```diff
@@ -179,6 +179,17 @@ const Model = types
 
+      onDispose() {
+        self.regs.forEach((region) => {
+          try {
```
do we expect errors there so we have to catch them?

juliosgarbi replied:
it makes sense @hlomzik, I was trying to make things safer, but seeing it up close there is no reason for this validation, thanks! :)

bmartel commented:
I have the same question as @hlomzik but outside of that, the rest looks good. Approving based on the resolution of that comment.

juliosgarbi committed:
remove the validation of errors on clearSpans

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/9194494742

juliosgarbi committed:
add semi-colon

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/9197947274
